src/ErrorProne.NET.Core/CompilationExtensions.cs

```diff
@@ -124,30 +138,29 @@ public static (INamedTypeSymbol? taskType, INamedTypeSymbol? taskOfTType, INamed
             return (taskType, taskOfTType, valueTaskOfTType);
         }
 
-        public static bool IsTaskLike(this ITypeSymbol? returnType, Compilation compilation)
+        public static bool IsTaskLike(this ITypeSymbol? returnType, TaskTypesInfo info)
         {
             if (returnType == null)
             {
                 return false;
             }
 
-            var (taskType, taskOfTType, valueTaskOfTType) = GetTaskTypes(compilation);
```

Comment:
This code path was repeatedly called, and `GetTaskTypes` would call `compilation.GetTypeByMetadataName`. For performance, it's recommended to do this call once, in compilation start (i.e, using `RegisterCompilationStartAction`).

So, instead of passing `Compilation` here, the symbols calculated in compilation start should be passed. Then, I needed to "group" all task symbols in `TaskTypesInfo` instead of passing multiple parameters here.